### PR TITLE
feat: Add support for Boba network

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
This commit adds support for the Boba network with the following changes:

Added assets for the Boba network located at 'src/assets/'
Added a feed selector for the Boba network located at 'src/components/FeeSelector'
Added a chain selector for the Boba network located at 'src/components/NavBar/ChainSelector.tsx'
Added a network component for the Boba network located at 'src/components/NetworkAlert/NetworkAlert.tsx'
Added addresses for [FACTORYU, MULTICAL, V3, ROUTER, QUOTER, NON FUNGIBLE ASSET MANAGER, TICKLENS] on the Boba network located at 'src/constants/addresses.ts'
Added Boba network types and hooks located at 'src/graphql/data/generated/types-and-hooks.ts'
Added the graphql chain id for the Boba network located at 'src/graphql/data/util.tsx'
Added the Boba network chain information located at 'src/constants/chainInfo.ts'
Added the Boba network graphql api url located at 'src/graphql/thegraph/apollo.ts'
Added gas override for the Boba network in 'src/hooks/useClientSideV3Trade.ts'
Added support for the Boba network in 'src/lib/hooks/useCurrencyLogoURIs.ts'
Added the Boba network color theme located at 'src/theme/components/RadialGradientByChainUpdater.ts'
Added the Boba network explorer located at 'src/utils/getExplorerLink.test.ts'
Added the Boba network explorer test located at 'src/utils/getExplorerLink.ts'
Added native token support for the Boba network located at 'src/utils/nativeTokens.ts'

### CODE REVIEW

**Comment by alvaro-ricotta** [comment]
Getting this error : 

image

I´m trying to disable Universal Router for your chain

File: src/constants/chains.ts [hunk 1]
```diff
@@ -30,6 +32,7 @@ export const CHAIN_IDS_TO_NAMES = {
   [SupportedChainId.ARBITRUM_GOERLI]: 'arbitrum_goerli',
   [SupportedChainId.OPTIMISM]: 'optimism',
   [SupportedChainId.OPTIMISM_GOERLI]: 'optimism_goerli',
+  [SupportedChainId.BOBA]: 'boba',
 }
```
Comment (souradeep-das) [inline src/constants/chains.ts:43]: lets add support for BOBA_GOERLI too to the interface (for the testnet)

(although, i'm not sure what the future of goerli is - but its what we have now, we could probably replace it when we change our testnet)

File: src/constants/lists.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ const WRAPPED_LIST = 'wrapped.tokensoft.eth'
 export const OPTIMISM_LIST = 'https://static.optimism.io/optimism.tokenlist.json'
 export const ARBITRUM_LIST = 'https://bridge.arbitrum.io/token-list-42161.json'
 export const CELO_LIST = 'https://celo-org.github.io/celo-token-list/celo.tokenlist.json'
+export const BOBA_LIST = 'https://raw.githubusercontent.com/bobanetwork/token-list/main/boba.tokenlist.json'
```
Comment (souradeep-das) [inline src/constants/lists.ts]: we could also use this github pages url instead?
https://bobanetwork.github.io/token-list/boba.tokenlist.json

File: src/constants/networks.ts [hunk 3]
```diff
@@ -73,6 +73,10 @@ export const FALLBACK_URLS: { [key in SupportedChainId]: string[] } = {
     // "Safe" URLs
     `https://alfajores-forno.celo-testnet.org`,
   ],
+  [SupportedChainId.BOBA]: [
+    // "Safe" URLs
+    `https://lightning-replica.boba.network/`,
```
Comment (souradeep-das) [inline src/constants/networks.ts:78]: should we use mainnet.boba.network instead? cc @InoMurko

File: src/constants/tokens.ts [hunk 4]
```diff
@@ -374,6 +437,19 @@ function getCeloNativeCurrency(chainId: number) {
   }
 }
 
+export function isBoba(chainId: number): chainId is SupportedChainId.BOBA {
+  return chainId === SupportedChainId.BOBA
+}
+
+function getBobaNativeCurrency(chainId: number) {
```
Comment (souradeep-das) [inline src/constants/tokens.ts:444]: Boba's default native currency is $ETH, and $BOBA is the alt fee token
Is there any fee switching logic involved/required?

File: src/constants/tokens.ts [hunk 5]
```diff
@@ -357,6 +413,13 @@ export const WRAPPED_NATIVE_CURRENCY: { [chainId: number]: Token | undefined } =
     'CELO',
     'Celo native asset'
   ),
+  [SupportedChainId.BOBA]: new Token(
+    SupportedChainId.BOBA,
+    '0x4200000000000000000000000000000000000006',
+    18,
+    'WETH',
```
Comment (souradeep-das) [inline src/constants/tokens.ts]: this is the ETH token on L2,
wrapped Ether on L2 is at '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000'

can we check how optimism handles the native token on interface? (note: the native token acts similar, however the difference being optimism has WETH on 0x42...6 and ETH on 0xDead..000
boba has WETH on 0xDead..000 and ETH on 0x42...6)

File: src/constants/tokens.ts [hunk 6]
```diff
@@ -374,6 +437,19 @@ function getCeloNativeCurrency(chainId: number) {
...
 }
 
+export function isBoba(chainId: number): chainId is SupportedChainId.BOBA {
+  return chainId === SupportedChainId.BOBA
+}
+
+function getBobaNativeCurrency(chainId: number) {
+  switch (chainId) {
+    case SupportedChainId.BOBA:
+      return BOBA_BOBA
```
Comment (souradeep-das) [inline src/constants/tokens.ts]: Boba's default native currency is $ETH, and $BOBA is the alt fee token
Is there any fee switching logic involved/required?

File: src/constants/networks.ts (hunk 3 above)
Comment (InoMurko) [reply src/constants/networks.ts:78]: it's a list, let's add all! :-)

**Comment by InoMurko** [comment]
> Getting this error :
> 
> image
> 
> I´m trying to disable Universal Router for our chain

If I remember correctly, Universal Router prerequisite is Uniswap v2, right @souradeep-das ?

File: src/constants/chains.ts (hunk 1 above)
Comment (alvaro-ricotta) [reply src/constants/chains.ts:43]: Goerli has been added

File: src/constants/lists.ts (hunk 2 above)
Comment (alvaro-ricotta) [reply src/constants/lists.ts]: URL has been updated

File: src/constants/networks.ts (hunk 3 above)
Comment (alvaro-ricotta) [reply src/constants/networks.ts:78]: Mainnet rpc has been added

File: src/constants/tokens.ts (hunk 5 above)
Comment (alvaro-ricotta) [reply src/constants/tokens.ts]: I already update this for Eth into our network, also i double check the weth url, it´s correct now both both networks (Goerli/Boba)

File: src/constants/tokens.ts (hunk 6 above)
Comment (alvaro-ricotta) [reply src/constants/tokens.ts]: hey @souradeep-das it´s ETH ? not wETH? in that case i already update the token address , i´ll be reviewing if there exist a fee-switching review

Comment (souradeep-das) [reply src/constants/tokens.ts]: awesome, yes! the native currency is ETH (0x4200000000000000000000000000000000000006)

**Comment by souradeep-das** [comment]
> If I remember correctly, Universal Router prerequisite is Uniswap v2, right @souradeep-das ?

yes, thats what we found! let me double check this though

**Comment by souradeep-das** [comment]
looks like they will make universal router the default moving forward. other chains without v2 seems to have deployed with addr(0) for v2 - will deploy it then

**Comment by InoMurko** [comment]
the last blocker here is as it seems the universal router, right?

File: src/constants/addresses.ts [hunk 7]
```diff
@@ -37,6 +37,20 @@ const ARBITRUM_GOERLI_QUOTER_ADDRESSES = '0x1dd92b83591781D0C6d98d07391eea4b9a60
 const ARBITRUM_GOERLI_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES = '0x622e4726a167799826d1E1D150b076A7725f5D81'
 const ARBITRUM_GOERLI_TICK_LENS_ADDRESSES = '0xb52429333da969a0C79a60930a4Bf0020E5D1DE8'
 
+const BOBA_V3_CORE_FACTORY_ADDRESSES = '0xFFCd7Aed9C627E82A765c3247d562239507f6f1B'
+const BOBA_V3_MIGRATOR_ADDRESSES = '0x9ab1E646877732DD76bd662DC242723D7e4B42ba'
+const BOBA_MULTICALL_ADDRESS = '0xd213Ff319F8d28Fe31EbDB75b23aA61a2b905320'
+const BOBA_QUOTER_ADDRESSES = '0x1c1fcecced3241b17c80e8b1e52a6ee0ea372fe1'
```
Comment (souradeep-das) [inline src/constants/addresses.ts]: I think this is supposed to be the 'QuoterV2' address instead?

File: src/constants/chainInfo.ts [hunk 8]
```diff
@@ -188,6 +190,32 @@ const CHAIN_INFO: ChainInfoMap = {
...
+    label: 'Boba',
+    logoUrl: bobaLogo,
+    circleLogoUrl: BobaNetworkLogoUrl,
+    nativeCurrency: { name: 'Boba Network', symbol: 'BOBA', decimals: 18 },
+    defaultListUrl: BOBA_LIST,
+  },
+  [SupportedChainId.BOBA_GOERLI]: {
+    networkType: NetworkType.L2,
+    blockWaitMsBeforeWarning: ms`10m`,
+    bridge: 'https://goerli.boba.network',
```
Comment (souradeep-das) [inline src/constants/chainInfo.ts]:
```suggestion
    bridge: 'https://gateway.boba.network/,
```

File: src/constants/chainInfo.ts [hunk 9]
```diff
@@ -188,6 +190,32 @@ const CHAIN_INFO: ChainInfoMap = {
...
+    networkType: NetworkType.L2,
+    blockWaitMsBeforeWarning: ms`10m`,
+    bridge: 'https://gateway.boba.network/',
+    docs: 'https://docs.boba.network/',
+    explorer: 'https://bobascan.com/',
+    infoLink: 'https://info.uniswap.org/#/boba',
+    label: 'Boba',
+    logoUrl: bobaLogo,
+    circleLogoUrl: BobaNetworkLogoUrl,
+    nativeCurrency: { name: 'Boba Network', symbol: 'BOBA', decimals: 18 },
```
Comment (souradeep-das) [inline src/constants/chainInfo.ts]: nativeCurrency is Ether

File: src/constants/chainInfo.ts [hunk 10]
```diff
@@ -188,6 +190,32 @@ const CHAIN_INFO: ChainInfoMap = {
...
+    networkType: NetworkType.L2,
+    blockWaitMsBeforeWarning: ms`10m`,
+    bridge: 'https://goerli.boba.network',
+    docs: 'https://docs.boba.network/',
+    explorer: 'https://testnet.bobascan.com/',
+    infoLink: 'https://info.uniswap.org/#/boba',
+    label: 'Boba Goerli',
+    logoUrl: bobaLogo,
+    circleLogoUrl: BobaNetworkLogoUrl,
+    nativeCurrency: { name: 'Boba Network', symbol: 'BOBA', decimals: 18 },
```
Comment (souradeep-das) [inline src/constants/chainInfo.ts]: native currency is Goerli Ether

File: src/constants/chains.ts [hunk 11]
```diff
@@ -34,6 +37,8 @@ export const CHAIN_IDS_TO_NAMES = {
   [SupportedChainId.ARBITRUM_GOERLI]: 'arbitrum_goerli',
   [SupportedChainId.OPTIMISM]: 'optimism',
   [SupportedChainId.OPTIMISM_GOERLI]: 'optimism_goerli',
+  [SupportedChainId.BOBA]: 'boba',
+  [SupportedChainId.BOBA_GOERLI]: 'boba_goerli',
 }
 
 /**
```
Comment (souradeep-das) [inline src/constants/chains.ts:44]: L55 - wondering if we need to add to SUPPORTED_GAS_ESTIMATE_CHAIN_IDS and UNSUPPORTED_V2POOL_CHAIN_IDS?

File: src/constants/routing.ts [hunk 12]
```diff
@@ -2,21 +2,26 @@
 import { Currency, Token } from '@uniswap/sdk-core'
 import { SupportedChainId } from 'constants/chains'
 
+import { USDC_BOBA_GOERLI } from './tokens'
```
Comment (souradeep-das) [inline src/constants/routing.ts:5]: squeeze this inside the next import?

File: src/constants/tokens.ts [hunk 13]
```diff
@@ -357,6 +454,21 @@ export const WRAPPED_NATIVE_CURRENCY: { [chainId: number]: Token | undefined } =
     'CELO',
     'Celo native asset'
   ),
+
+  [SupportedChainId.BOBA]: new Token(
+    SupportedChainId.BOBA,
+    '0x4200000000000000000000000000000000000006',
```
Comment (souradeep-das) [inline src/constants/tokens.ts:460]: i think this one is asking for the wrapped Ether currency?
If this is - the wrapped Ether token is at '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000'

can we update this and Boba_goerli?

File: src/constants/tokens.ts [hunk 14]
```diff
@@ -294,6 +310,87 @@ export const CEUR_CELO_ALFAJORES = new Token(
...
+
+export const LINK_BOBA = new Token(
+  SupportedChainId.BOBA,
+  '0xD5D5030831eE83e22a2C9a5cF99931A50c676433',
+  18,
+  'LINK',
+  'ChainLink Token'
+)
+
+const ETH_BOBA = new Token(SupportedChainId.BOBA, '0x4200000000000000000000000000000000000006', 18, 'ETH', 'Ether')
```
Comment (souradeep-das) [inline src/constants/tokens.ts:369]: minor: lint

File: src/hooks/useClientSideV3Trade.ts [hunk 15]
```diff
@@ -18,6 +18,8 @@ const QUOTE_GAS_OVERRIDES: { [chainId: number]: number } = {
   [SupportedChainId.CELO_ALFAJORES]: 50_000_000,
   [SupportedChainId.POLYGON]: 40_000_000,
   [SupportedChainId.POLYGON_MUMBAI]: 40_000_000,
+  [SupportedChainId.BOBA]: 50_000_000,
```
Comment (souradeep-das) [inline src/hooks/useClientSideV3Trade.ts:21]: not sure about this gasOverride, is there a way to test this? 🤔

File: src/lib/hooks/useCurrencyLogoURIs.ts [hunk 16]
```diff
@@ -6,7 +6,7 @@ import { isAddress } from 'utils'
 import EthereumLogo from '../../assets/images/ethereum-logo.png'
 import CeloLogo from '../../assets/svg/celo_logo.svg'
 import MaticLogo from '../../assets/svg/matic-token-icon.svg'
-import { isCelo, NATIVE_CHAIN_ID, nativeOnChain } from '../../constants/tokens'
+import { isBoba, isCelo, NATIVE_CHAIN_ID, nativeOnChain } from '../../constants/tokens'
 
 type Network = 'ethereum' | 'arbitrum' | 'optimism' | 'polygon'
```
Comment (souradeep-das) [inline src/lib/hooks/useCurrencyLogoURIs.ts]: L13 do we need to add to this? 
`chainIdToNetworkName`

File: src/utils/nativeTokens.ts [hunk 17]
```diff
@@ -9,6 +9,7 @@ export function getNativeTokenDBAddress(chain: Chain): string | undefined {
   }
   switch (chain) {
     case Chain.Celo:
+    case Chain.Boba:
```
Comment (souradeep-das) [inline src/utils/nativeTokens.ts]: why are we returning the wrapped address for boba?

File: src/constants/addresses.ts [hunk 18]
```diff
@@ -129,6 +151,8 @@ export const NONFUNGIBLE_POSITION_MANAGER_ADDRESSES: AddressMap = {
   [SupportedChainId.CELO_ALFAJORES]: CELO_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES,
   [SupportedChainId.OPTIMISM_GOERLI]: OPTIMISM_GOERLI_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES,
   [SupportedChainId.ARBITRUM_GOERLI]: ARBITRUM_GOERLI_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES,
+  [SupportedChainId.BOBA]: BOBA_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES,
+  [SupportedChainId.BOBA_GOERLI]: BOBA_GOERLI_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES,
 }
 
 export const ENS_REGISTRAR_ADDRESSES: AddressMap = {
```
Comment (souradeep-das) [inline src/constants/addresses.ts:158]: L167: lets add TICK_LENS_ADDRESSES for Boba and boba goerli

File: src/constants/addresses.ts (hunk 7 above)
Comment (alvaro-ricotta) [reply src/constants/addresses.ts]: Hey, @souradeep-das this list is standard. 

image

Comment (souradeep-das) [reply src/constants/addresses.ts]: yes the BOBA_QUOTER_ADDRESSES is pointing to the wrong address I believe

File: src/constants/chainInfo.ts (hunk 9 above)
Comment (alvaro-ricotta) [reply src/constants/chainInfo.ts]: Native token updated

File: src/constants/chains.ts (hunk 11 above)
Comment (alvaro-ricotta) [reply src/constants/chains.ts:44]: Unsuported v2 , and supported gast estimated added for both networks

File: src/constants/tokens.ts (hunk 14 above)
Comment (alvaro-ricotta) [reply src/constants/tokens.ts:369]: This is automatically Lint like this one, same case for CELO

File: src/constants/chainInfo.ts (hunk 10 above)
Comment (alvaro-ricotta) [reply src/constants/chainInfo.ts]: Native currency updated on gorli too

File: src/constants/addresses.ts (hunk 18 above)
Comment (alvaro-ricotta) [reply src/constants/addresses.ts:158]: It´s there.

File: src/hooks/useClientSideV3Trade.ts (hunk 15 above)
Comment (alvaro-ricotta) [reply src/hooks/useClientSideV3Trade.ts:21]: Based on @Boyuan-Chen answer, 50_000_000 is the right number

File: src/constants/addresses.ts (hunk 7 above)
Comment (alvaro-ricotta) [reply src/constants/addresses.ts]: quoter routers updated

File: src/utils/nativeTokens.ts (hunk 17 above)
Comment (alvaro-ricotta) [reply src/utils/nativeTokens.ts]: Fixed

File: src/hooks/useClientSideV3Trade.ts (hunk 15 above)
Comment (boyuan-chen) [reply src/hooks/useClientSideV3Trade.ts:21]: It's the maximum gas for `eth_call`.

**Comment by boyuan-chen** [comment]
Should we avoid the code style changes? Does Uniswap have a template for the code style?

**Comment by InoMurko** [comment]
> Should we avoid the code style changes? Does Uniswap have a template for the code style?

We'll see what the feedback is. Yes in general we should.

File: src/hooks/useClientSideV3Trade.ts (hunk 15 above)
Comment (souradeep-das) [reply src/hooks/useClientSideV3Trade.ts:21]: will that make it over the block gas limit though?

File: src/lib/hooks/useCurrencyLogoURIs.ts [hunk 19]
```diff
@@ -18,6 +18,10 @@ export function chainIdToNetworkName(networkId: SupportedChainId): Network {
       return 'arbitrum'
     case SupportedChainId.OPTIMISM:
       return 'optimism'
+    case SupportedChainId.BOBA:
+      return 'boba'
+    case SupportedChainId.BOBA_GOERLI:
+      return 'boba_goerli'
     case SupportedChainId.POLYGON:
       return 'polygon'
     default:
```
Comment (souradeep-das) [inline src/lib/hooks/useCurrencyLogoURIs.ts:27]: aah i think getTokenLogoURI would still require the `isBoba()` condition to fetch the correct link?

**Comment by souradeep-das** [comment]
was able to spin up locally, looks good to me!
can we also address these two remaining comments plz?

https://github.com/bobanetwork/interface/pull/2#discussion_r1132266316
https://github.com/bobanetwork/interface/pull/2#discussion_r1133622252

**Pull request closed without merging**